### DISCUSSION TITLE
how to do without the id but another primary key

### DISCUSSION THREAD

**Discussion by Unisolgr14** [Power-Components/livewire-powergrid#613] [Q&A] [answered]
I have my tables without id but with like member_id as an example. so without this id filed iam unable to use the powergrid as id is use in the order by clause.how do I use poweergrid without the id filed?

**Comment by dansysanalyst** [comment, marked as answer]
You can define as follows:

```php

final class RestaurantsTable extends PowerGridComponent
{
    use ActionButton;

    public string $primaryKey = 'restaurants.custom_id';

    public string $sortField = 'restaurants.custom_id';

     //...
```

👉 https://livewire-powergrid.com/#/table/datasource?id=custom-primary-key

**Reply by Unisolgr14** [reply]
A hundred apologies for my question. I do not know how I missed this in the documentation.I have already read before, now I will go through this again.
 Thank you so much for your reply.

**Reply by dansysanalyst** [reply]
Thank you for your question! 

Don't worry, you are not going crazy 😁

I made a PR into the doc after you asked because I thought it was not clear enough.
